pkg/liveshare/client.go

```diff
@@ -112,15 +119,49 @@ func Connect(ctx context.Context, opts Options) (*Session, error) {
 	s := &Session{
 		ssh:             ssh,
 		rpc:             rpc,
+		grpc:            grpc.New(),
 		clientName:      opts.ClientName,
 		keepAliveReason: make(chan string, 1),
 		logger:          opts.Logger,
 	}
 	go s.heartbeat(ctx, 1*time.Minute)
 
+	// Connect to the gRPC server so we can make requests anywhere we have access to the session
+	s.connectToGrpcServer(ctx, opts.SessionToken)
+
 	return s, nil
 }
 
+// Connects to the gRPC server running on the host VM
+func (s *Session) connectToGrpcServer(ctx context.Context, token string) error {
+	listen, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", 0))
```

Comment:
This listener is never closed, what is the expected lifetime for this connection?

Reply:
This connection should be open as long as the Live Share session is connected.

Reply:
In this case, I wonder if this connect logic shouldn't go to the `grpc.Client` connect method, and have `grpc.Client` expose a `Close() error` method similar to the RPC client, and then we can call `grpc.Close()` inside `session.Close`: https://github.com/cli/cli/blob/0b2348aa610e95435044d35b7e19289264f3a888/pkg/liveshare/session.go#L38

Something to look into:
1. Can we call `Close()` on a `*grpc.ClientConn`?
2. If we can, what does that close?
3. I doubt it closes our local listener, which if it doesn't we'll need to write some code to manage the lifetime of the listener as we need to clean up resources once the session is closed

Reply:
That's where I originally had it but with this new way of including the gRPC client in the LS session would introduce a circular dependency since the port forwarder is in the `liveshare` package.

1. We can.
2. From the description, it says `Close tears down the ClientConn and all underlying connections.`.
3. Not sure if it does either but I can test it. I could also send the `net.Listener` through the `grpc.Connect` call and then store that to be closed in `grpc.Close` later to be safe?

Reply:
Yeah looks like `*grpc.ClientConn` doesn't close the local listener so I'll go ahead and do what I proposed.